protocol/contracts/governance/ElderElection.sol

```diff
@@ -0,0 +1,149 @@
+     */
+    function nominate(uint256 discordId) external onlyRole(CAN_NOMINATE) {
+        templars.checkExists(discordId);
+
+        if (!candidates[discordId]) {
+          candidates[discordId] = true;
+          numCandidates += 1;
+          emit UpdateNomination(discordId, true);
+        }
+    }
+
+    /**
+     * @notice Remove the given discord ID as a candidate
+     */
+    function resign(uint256 discordId) external onlyRole(CAN_NOMINATE) {
+        templars.checkExists(discordId);
+
+        if (candidates[discordId]) {
+          candidates[discordId] = false;
+          numCandidates -= 1;
+          emit UpdateNomination(discordId, false);
+        }
+    }
+
+    function setEndorsements(uint256[] memory discordIds) external {
+        _setEndorsements(msg.sender, discordIds);
+    }
+
+    function _setEndorsements(address account, uint256[] memory discordIds) internal {
+        if (discordIds.length > numCandidates) {
+          revert TooManyEndorsements();
+        }
+
+        endorsementsBy[account] = discordIds;
+        emit UpdateEndorsements(account, discordIds);
+    }
+
+    /**
+     * @notice Toggle the 
+     * (assuming the owner has given authority for the caller to act on their behalf)
+     *
+     * @dev amount is explicit, to allow use case of partial vault withdrawals
+     */
+    // TODO(butlerji): Need to test this works. The idea is I'd sign a message on mainnet, which gets sent to our backend, then sent to 
+    //                 the mempool on whatever L2 we are running this on.
+    function relayedSetEndorsementsFor(address account, uint256[] memory discordIds, uint256 deadline, bytes calldata signature) external {
+        if (block.timestamp > deadline) revert DeadlineExpired(deadline);
+
+        bytes32 structHash = keccak256(abi.encode(TOGGLE_ENDORSEMENTS_FOR_TYPEHASH, account, discordIds, deadline, _useNonce(account)));
+        bytes32 digest = _hashTypedDataV4(structHash);
+        (address signer, ECDSA.RecoverError err) = ECDSA.tryRecover(digest, signature);
+        if (err != ECDSA.RecoverError.NoError) {
+            revert InvalidSignature(account);
+        }
+        if (signer != account) revert InvalidSignature(account);
+
+        _setEndorsements(account, discordIds);
+    }
+
+    /**
+    * See {IERC20Permit-DOMAIN_SEPARATOR}.
+    */
+    // solhint-disable-next-line func-name-mixedcase
+    function DOMAIN_SEPARATOR() external view returns (bytes32) {
+        return _domainSeparatorV4();
+    }
+
+    /**
+     * "Consume a nonce": return the current value and increment.
+     */
+    function _useNonce(address _owner) internal returns (uint256 current) {
+        Counters.Counter storage nonce = _nonces[_owner];
+        current = nonce.current();
+        nonce.increment();
+    }
+
+    event UpdateNomination(uint256 discordId, bool isNominated);
+    event UpdateEndorsements(address account, uint256[] discordId);
```

Comment:
Should add `indexed` keyword to the params that are likely to be searched on. Basically allows those params to be used as filters.